### DISCUSSION TITLE
How to set expiry time of message sent to device?

### DISCUSSION THREAD

**Discussion by arathod-eic** [Azure/azure-iot-sdk-java#1703] [Q&A] [answered]
I am trying to upgrade to the 2.X.X releases from the 1.34.2 SDK version.

In the previous release, a method was available to set the expiry of a message sent to the device. 

As per my understanding, the package has been changed to **com.microsoft.azure.sdk.iot.service.messaging.Message** in 2.1.5 from **com.microsoft.azure.sdk.iot.service.Message** in 1.34.2 for the Message class.

Method name in v1.34.2 in Message Class : setExpiryTimeUtc(long milliseconds)

Is there any new method of setting an expiry to the message?

Appreciate the help. Thanks.

**Comment by timtay-microsoft** [comment, marked as answer]
It looks like this functionality was lost in the upgrade to v2. We've migrated the code/issues for the IoT hub service SDK over [here](https://github.com/azure/azure-iot-service-sdk-java), so can you open a Github issue thread over there?

**Reply by arathod-eic** [reply]
Thank you for the quick response - I have opened an issue in the repo that you have mentioned.
